kernel-spark/src/main/java/io/delta/kernel/spark/utils/RowTrackingUtils.java

```diff
@@ -0,0 +1,170 @@
+  private static final String BASE_ROW_ID = "base_row_id";
+  private static final String ROW_ID = "row_id";
+  private static final String DEFAULT_ROW_COMMIT_VERSION = "default_row_commit_version";
+  private static final String ROW_COMMIT_VERSION = "row_commit_version";
+
+  // Metadata keys for row tracking metadata fields
+  private static final String BASE_ROW_ID_METADATA_COL_ATTR_KEY = "__base_row_id_metadata_col";
+  private static final String DEFAULT_ROW_COMMIT_VERSION_METADATA_COL_ATTR_KEY =
+      "__default_row_version_metadata_col";
+  private static final String ROW_ID_METADATA_COL_ATTR_KEY = "__row_id_metadata_col";
+  private static final String ROW_COMMIT_VERSION_METADATA_COL_ATTR_KEY =
+      "__row_commit_version_metadata_col";
+
+  private RowTrackingUtils() {}
+
+  /**
+   * Check if row tracking is enabled for reading.
+   *
+   * @param protocol the protocol to check
+   * @param metadata the metadata to check
+   * @return true if row tracking is enabled
+   * @throws IllegalStateException if row tracking is enabled in metadata but not supported by
+   *     protocol
+   */
+  public static boolean isEnabled(Protocol protocol, Metadata metadata) {
+    boolean isEnabled = TableConfig.ROW_TRACKING_ENABLED.fromMetadata(metadata);
+    if (isEnabled && !TableFeatures.isRowTrackingSupported(protocol)) {
+      throw new IllegalStateException(
+          "Table property 'delta.enableRowTracking' is set on the table but this table version "
+              + "doesn't support table feature 'delta.feature.rowTracking'.");
+    }
+    return isEnabled;
+  }
+
+  /**
+   * Create the row tracking metadata struct fields for reading.
+   *
+   * <p>The order and presence of fields matches Spark V1 implementation:
+   *
+   * <ul>
+   *   <li>row_id (generated field, only if delta.rowTracking.materializedRowCommitVersionColumnName
+   *       is configured)
+   *   <li>base_row_id (constant field, always present when row tracking is enabled)
+   *   <li>default_row_commit_version (constant field, always present when row tracking is enabled)
+   *   <li>row_commit_version (generated field, only if materialized column name is configured)
+   * </ul>
+   *
+   * @param protocol the protocol
+   * @param metadata the metadata
+   * @param nullableConstantFields whether constant fields should be nullable
+   * @param nullableGeneratedFields whether generated fields should be nullable
+   * @return list of struct fields for row tracking metadata, or empty list if row tracking is not
+   *     enabled
+   */
+  public static List<StructField> createMetadataStructFields(
+      Protocol protocol,
+      Metadata metadata,
+      boolean nullableConstantFields,
+      boolean nullableGeneratedFields) {
+    if (!isEnabled(protocol, metadata)) {
+      return new ArrayList<>();
+    }
+
+    List<StructField> fields = new ArrayList<>();
+
+    // Add row_id (generated field) if materialized column name is configured
```

Comment:
You should throw an exception if the materialized column name is missing.
When row tracking is enabled, materialized column **must** be assigned.

Similar to https://github.com/delta-io/delta/blob/8e72700b497caa82ddd14daf2b4f6258cd7e25e6/kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java#L112 (it would be good to define a common exception that can be reused)